src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java

```diff
@@ -53,24 +49,27 @@
   private Table table;
 
   @Override
-  public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
-    String tableDir = StringUtils.substringAfter(job.get("location"), "file:");
-
-    //Change this to use whichever Catalog the table was made with i.e. HiveCatalog instead etc.
+  public InputSplit[] getSplits(JobConf job, int numSplits) {
+    //TODO: Change this to use whichever Catalog the table was made with i.e. HiveCatalog instead etc.
     HadoopTables tables = new HadoopTables(job);
-    table = tables.load(tableDir);
-    TableScan scan = table.newScan();
+    String tableDir = job.get("location");
+
+    URI location = null;
+    try {
+      location = new URI(tableDir);
+    } catch (URISyntaxException e) {
+      LOG.error("Unable to create URI for table location: '" + tableDir + "'");
```

Comment:
Shouldn't we throw this instead of logging it? Perhaps as an UncheckedIOException? It not you'll get a NPE on line 63 which will have less information.

Reply:
Yeah I agree it would be best to throw an exception - got thrown off by the interface restrictions not letting me throw the URISyntaxException in the method signature :L I can throw an IOException too, what would be the reasoning behind using either UncheckedIOException or just IOException?

Reply:
Try both and you'll see ;)